apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -63,3 +80,33 @@ export const WarningLabel = ({children}) => (
 WarningLabel.propTypes = {
   children: React.PropTypes.node
 };
+
+/**
+ * Given a search query, generate a results list of animationProps objects that
+ * can be displayed and used to add an animation to the project.
+ * @param {string} searchQuery - text entered by the user to find an animation
+ */
+function searchAnimations(searchQuery) {
+  // Make sure to generate the search regex in advance, only once.
+  // Search is case-insensitive
+  // Match any word boundary or underscore followed by the search query.
+  // Example: searchQuery "bar"
+  //   Will match: "barbell", "foo-bar", "foo_bar" or "foo bar"
+  //   Will not match: "foobar", "ubar"
+  const searchRegExp = new RegExp('(?:\\b|_)' + searchQuery, 'i');
```

Comment:
TODO: I suppose I should escape `searchQuery` somehow?  I mean, it's pretty cool to support regex in our search, but probably not helpful to students.